packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -30,6 +64,12 @@ const userSignUp = z
 			})
 			.max(UserValidationRule.PASSWORD_MAX_LENGTH, {
 				message: UserValidationMessage.PASSWORD_MAX_LENGTH,
+			})
+			.regex(/^[\d!#$@A-Za-z]*$/, {
+				message: UserValidationMessage.PASSWORD_INVALID_CHARACTERS,
+			})
+			.regex(/^(?=.*[A-Za-z])(?=.*\d)/, {
+				message: UserValidationMessage.PASSWORD_REQUIRES_LETTER_AND_NUMBER,
```

Comment:
Suggestion to create such constants and reuse them in different forms.

Reply:
Agree. Let create a Regex enum and give each regexp a meaningful name